lib/honeybadger.ex

```diff
@@ -209,15 +210,34 @@ defmodule Honeybadger do
     |> Client.send_notice()
   end
 
+  @doc """
+  Retrieves the context that will get sent to the Honeybadger API when/if an
+  exception occurs in the current process.
+  """
+  @spec context :: context
   def context do
     (Process.get(@context) || %{}) |> Enum.into(Map.new)
   end
 
-  def context(keyword_or_map) do
-    Process.put(@context, Map.merge(context(), Enum.into(keyword_or_map, %{})))
+  @doc """
+  Merges `additional_context` into the the context that will get sent to the
+  Honeybadger API when/if an exception occurs in the current process.
+  """
+  @spec context(map | keyword) :: context
+  def context(additional_context) do
```

Comment:
I like the improved variable name here. What do you think about adding a guard, as this will blow up without a map or a keyword list? Also, it cements that the way to clear the context would be with `clear_context/0` rather than `context(nil)`.